### PR TITLE
core: fix wide string handling in html::text_format

### LINKED ISSUE #5839: Hebrew characters are not displayed properly.
### Describe the bug

Hebrew embedded text, which used to appear correctly (although, with characters in incorrect order: #3921), appear to no longer display properly.

When hebrew text is used, rather than drawing the hebrew letters, a combination of incorrect characters appears.

Here is a screenshot showing [clock.swf](https://github.com/ruffle-rs/ruffle/files/6273851/clock.zip) on Flash Player 29, and the regression between an old version of Nightly 2021-04-07 (which displayed hebrew characters), and current version Nightly 2021-12-10 (which displays incorrect characters):

image

Link to SWF: [clock.zip](https://github.com/ruffle-rs/ruffle/files/6273851/clock.zip)

### Expected behavior

Hebrew characters are displayed when hebrew text is embedded.

### Affected platform

Desktop app

### Operating system

Windows 10

### Browser

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by moulins**
Closes #5839.

### CODE REVIEW

File: core/src/html/text_format.rs [hunk 1]
```diff
@@ -575,11 +575,25 @@ impl FormatSpans {
...
+            ),
+        };
+
+        // Helper function to decode a byte sequence returned by quick_xml into a WString.
+        // TODO: use Cow<'_, WStr>?
+        let decode_to_wstr = |raw: &[u8]| -> WString {
+            if is_raw_latin1 {
+                WString::from_buf(raw.to_vec())
+            } else {
+                let utf8 = std::str::from_utf8(raw).expect("raw should be valid utf8");
```
Comment (relrelb) [inline core/src/html/text_format.rs:594]: Please use `str` instead of `std::str`.

File: core/src/html/text_format.rs [hunk 2]
```diff
@@ -737,8 +751,11 @@ impl FormatSpans {
                     format_stack.push(format);
                 }
                 Ok(Event::Text(e)) if !e.is_empty() => {
-                    let e = WString::from_buf(e.escaped().to_owned());
-                    let e = process_html_entity(&e).unwrap_or(e);
+                    let mut e = decode_to_wstr(e.escaped());
```
Comment (relrelb) [inline core/src/html/text_format.rs:754]: Why not keeping the `.unwrap_or()`?
```rs
let e = decode_to_wstr(e.escaped());
let e = process_html_entity(&e).unwrap_or(e);
```

File: core/src/html/text_format.rs (hunk 1 above)
Comment (moulins) [reply core/src/html/text_format.rs:594]: `from_utf8` isn't a method on `str`, it's a free function in `std::str`.

File: core/src/html/text_format.rs (hunk 2 above)
Comment (moulins) [reply core/src/html/text_format.rs:754]: No reason, I'll change it.

**Review by relrelb — APPROVED** [review]
LGTM

**Pull request merged**